core_backend/app/routers/question_answer.py

```diff
@@ -43,103 +45,108 @@ async def llm_response(
-            "Sorry, we are unable to understand your question. "
-            "Please rephrase your question and try again."
-        )
+@classify_safety
+@paraphrase_question
+@translate_question
+@identify_language
+async def get_llm_answer(
+    user_query_refined: UserQueryRefined,
+    response: UserQueryResponse,
+    qdrant_client: QdrantClient,
+) -> UserQueryResponse:
+    """
+    Get similar content and construct the LLM answer for the user query
+    """
+    if response.state == ResultState.ERROR:
         return response
-    if user_query_refined.original_language != IdentifiedLanguage.ENGLISH:
-        user_query_refined = translate_question(user_query_refined)
-    user_query_refined = paraphrase_question(user_query_refined)
-
-    # -- RAG call
     content_response = get_similar_content(
         user_query_refined, qdrant_client, int(QDRANT_N_TOP_SIMILAR)
     )
     response.content_response = content_response
     response.llm_response = get_llm_rag_answer(
         user_query_refined.query_text, content_response[0].response_text
     )
-    response.debug_info = {
-        "paraphrase": user_query_refined.query_text,
-        "original_language": user_query_refined.original_language,
-    }
-
-    await save_query_response_to_db(asession, user_query_db, response)
 
     return response
 
 
-@router.post("/embeddings-search")
-async def embeddings_search(
-    user_query: UserQueryBase,
-    asession: AsyncSession = Depends(get_async_session),
-    qdrant_client: QdrantClient = Depends(get_qdrant_client),
-) -> UserQueryResponse:
+async def get_user_query_and_response(
+    user_query: UserQueryBase, asession: AsyncSession
+) -> Tuple[UserQueryDB, UserQueryRefined, UserQueryResponse]:
     """
-    Embeddings search finds the most similar embeddings to the user query
-    from the vector db.
+    Get the user query from the request and save it to the db.
+    Construct an object for user query and a default response object.
     """
-
     feedback_secret_key = generate_secret_key()
-
     user_query_db = await save_user_query_to_db(
         asession, feedback_secret_key, user_query
     )
-
     user_query_refined = UserQueryRefined(
         **user_query.model_dump(), query_text_original=user_query.query_text
     )
-    user_query_refined = identify_language(user_query_refined)
-    if user_query_refined.original_language not in [
-        IdentifiedLanguage.UNKNOWN,
-        IdentifiedLanguage.ENGLISH,
-    ]:
-        user_query_refined = translate_question(user_query_refined)
-    user_query_refined = paraphrase_question(user_query_refined)
-
-    content_response = get_similar_content(
-        user_query, qdrant_client, int(QDRANT_N_TOP_SIMILAR)
-    )
-
     response = UserQueryResponse(
         query_id=user_query_db.query_id,
-        content_response=content_response,
-        llm_response=None,
+        content_response=None,
+        llm_response="Sorry, we cannot do that.",
```

Comment:
Correct me if I'm wrong but wouldn't this set the llm_response field of all embeddings-search queries to this? I guess we should think more generally about how to store queries for different endpoints but before we do that should we keep the default to None?

Reply:
You are totally right! Thanks for picking this up. Default should be None. I'll fix this.